app/controllers/pages_controller.rb

```diff
@@ -1,4 +1,9 @@
 class PagesController < ApplicationController
+  before_action :authenticate_user!, only: :dashboard
+
   def home
   end
+
+  def dashboard
+  end
```

Comment:
Essaye de penser "resourceful". Dans 99% des cas tu peux organiser tes routes avec `resources`/`resource`.

Il vaut mieux avoir beaucoup de petits controllers que quelques gros.

Le `only: :dashboard` ici est déjà un gros indice que l'action dashboard ne va pas être gérée comme l'action `home`.

Un bon article sur le sujet : http://jeromedalbert.com/how-dhh-organizes-his-rails-controllers/

Reply:
Et c'est vrai que je n'y avait pas pensé pour la home, mais pratiquement toutes les actions de controllers peuvent être du CRUD (index, show, new, create, edit, update, destroy).

La home pourrait être l'action `show` dans un `HomesController`.

Reply:
Mais du coup le dashboard tu en ferais quoi ?
Une show sur un DashboardsController ?
Voire même sous Users::DashboardController ?

Reply:
En fait si je comprends bien le mieux c'est de faire au maximum pour tout "CRUDer" ?

Reply:
`Users::DashboardsController` c'est pas déconnant (toujours au pluriel).

"tout CRUDer" ouais.

Reply:
Done dans un My::DashboardsController.

Reply:
`My` ? Ça veut dire quoi `My` ? `module My` c'est le module de qui ?

`module Users` c'est clair, c'est tout ce qui concerne les utilisateurs, pourquoi tu ne veux pas l'appeler `Users` ?

Reply:
Ah ok je pensais que tu connaissais le My. C'est censé être l'une des conventions de namespace pour ranger tout ce qui est relatif au current_user. Mais ça me va très bien Users aussi.

Reply:
Je ne suis pas sûr que ça soit une convention vraiment répandue dans la communauté rails; j'ai jamais croisé ça en 5 ans. :thinking: 

Sur nos sites on a des investisseurs, des entrepreneurs, des admins... Avec `My` ça ne serait pas clair.

Reply:
Ok c'est noté je ne le l'utiliserai plus alors.